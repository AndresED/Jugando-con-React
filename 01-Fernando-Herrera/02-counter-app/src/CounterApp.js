import PropTypes from 'prop-types';
import './index.css';
import React,{useState} from 'react';

const CounterApp = ({value}) =>{
    const [counter, setCounter] = useState(value);
    const handleAdd = ()=>{
        setCounter(counter+1);
    }
    const handleSubstration = ()=>{
        setCounter(counter-1);
    }
    const handleReset = ()=>{
        setCounter(value);
    }
    return (
        <>
        <h1>CounterApp</h1>
        <h2 id="contador">{counter}</h2>
        <button id="btnAdd" onClick={handleAdd}>+</button>
        <button id="btnReset" onClick={handleReset}>Reset</button>
        <button id="btnRemove" onClick={handleSubstration}>-</button>
        </>
    );
}
CounterApp.propTypes ={
    value: PropTypes.number.isRequired
}
export default CounterApp;