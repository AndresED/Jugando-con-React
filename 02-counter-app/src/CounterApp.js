import PropTypes from 'prop-types';
import './index.css';
import React,{useState} from 'react';

const CounterApp = ({value}) =>{
    const [counter, setCounter] = useState(value);
    const handleAdd = ()=>{
        setCounter(counter+1);
    }
    const handleSubstration = ()=>{
        setCounter(counter-1);
    }
    const handleReset = ()=>{
        setCounter(value);
    }
    return (
        <>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>
        <button onClick={handleAdd}>+</button>
        <button onClick={handleReset}>Reset</button>
        <button onClick={handleSubstration}>-</button>
        </>
    );
}
CounterApp.propTypes ={
    value: PropTypes.number.isRequired
}
export default CounterApp;