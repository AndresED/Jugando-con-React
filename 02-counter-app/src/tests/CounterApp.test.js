import React from 'react';
import CounterApp from '../CounterApp';
import '@testing-library/jest-dom';
import { shallow, mount, render } from 'enzyme';

describe('Pruebas sobre el componente CounterApp',()=>{
    test('Debe de generar un snatshop y mostrar sus valores por defecto',()=>{
        const wrapper = shallow(<CounterApp value={0} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar el valor por defecto de 100',()=>{
        const contador = 100;
        const wrapper = shallow(<CounterApp value={contador} />);
        expect(wrapper).toMatchSnapshot();
        const textCounter = wrapper.find("#contador").text();
        expect(textCounter).toBe(textCounter);
    })
})