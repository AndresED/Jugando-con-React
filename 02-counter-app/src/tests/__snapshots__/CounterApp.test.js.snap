// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente CounterApp Debe de generar un snatshop y mostrar sus valores por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    id="contador"
  >
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</Fragment>
`;

exports[`Pruebas sobre el componente CounterApp Debe de mostrar el valor por defecto de 100 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    id="contador"
  >
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</Fragment>
`;
